### PR TITLE
Add refresh token to oauth flows

### PR DESCRIPTION
**Pull request by dany-kun**

### CODE REVIEW

File: lib/oauth/oauth.ex [hunk 1]
```diff
@@ -24,4 +61,20 @@ defmodule OAuth do
...
+    Agent.get(agent(), get_fn)
+  end
+
+  defp store_in_cache(update_fn) do
+    Agent.update(agent(), update_fn)
+  end
+
+  # Use a local agent as a cache system
+  defp agent() do
+    case Agent.start_link(fn -> %{} end, name: __MODULE__) do
```
Comment (sabiwara) [inline lib/oauth/oauth.ex]: You typically should avoid starting "static" processes on the fly, this is the role the supervision tree 😉
If you use a named process, you don't need to even retrieve the `pid` each time, you can just use the process name [directly](https://hexdocs.pm/elixir/Agent.html#module-examples): 
```elixir
Agent.update(__MODULE__, update_fn)
```

Comment (sabiwara) [reply lib/oauth/oauth.ex]: Actually this cache might not work as you expect:
- calling start_link within the process handling a web request
- the agent is created and the two processes get linked: agent <=> request
- the request finished, taking down the request and the agent together
- the next request comes in: starts a new empty agent :p

Comment (dany-kun) [reply lib/oauth/oauth.ex]: > You typically should avoid starting "static" processes on the fly, this is the role the supervision tree wink If you use a named process, you don't need to even retrieve the `pid` each time, you can just use the process name [directly](https://hexdocs.pm/elixir/Agent.html#module-examples):
> 
> ```elixir
> Agent.update(__MODULE__, update_fn)
> ```

But I would still need to potentially create the agent before updating the state no?

Anyway, how would you create a on persistent cache system with a lifetime == my app process (or even shorter) ?

Comment (sabiwara) [reply lib/oauth/oauth.ex]: This is where the supervision tree comes in.
You can check the [plug doc](https://hexdocs.pm/plug/readme.html#supervised-handlers) to see how to add your plug server to your application tree, and then add your agent to the children.
More links:
- https://hexdocs.pm/elixir/Agent.html#module-how-to-supervise
- https://elixir-lang.org/getting-started/mix-otp/supervisor-and-application.html

Comment (dany-kun) [reply lib/oauth/oauth.ex]: OK. Updated/Gave it a try.

File: lib/oauth/oauth.ex [hunk 2]
```diff
@@ -25,3 +62,34 @@ defmodule OAuth do
...
+
+  def start(_type, _args) do
+    children = [
+      {OAuth.Cache, %{}}
+    ]
+
+    opts = [strategy: :one_for_one, name: Bootstrap.Supervisor]
+    Supervisor.start_link(children, opts)
+  end
+end
```
Comment (sabiwara) [inline lib/oauth/oauth.ex:95]: Nice, you got it working 👍
Just a note:tThis should typically be The application of your project, not related to oauth specifically.
It is typically stored in `lib/project_name/application.ex`, where you should start your webserver etc as well.
This is not blocking though, you can always refactor it later.

**Review by sabiwara — APPROVED** [review]
Nice, LGTM!

**Pull request merged**